developers/development/guidelines.md

```diff
@@ -14,73 +14,191 @@ Exceptions can certainly be made, but they should be discussed and approved by a
 Note that this list also serves as a checklist for code reviews on pull requests.
 To speed up the contribution process, we therefore advice to go through this checklist yourself before creating a pull request.
 
-## A. Code Style
-
-1. The [Java naming conventions](http://java.about.com/od/javasyntax/a/nameconventions.htm) should be used.
+## A. Directory and file layout
+
+The following directory and file layout must be respected:
+
+```
+.
++-- pom.xml  The buildsystem (maven) file that contains the version and name of your extension
+|
++-- NOTICE   license information
+|            3rd party content has to be given in the NOTICE file.
+|
++-- lib      3rd party jar files - Avoid to use those, use maven dependencies instead
+|
++-- src      Your source code
+|   +-- main/jave/...      The java sources
+|   +-- main/resources/... Other resource files like the binding.xml
+|   +-- test/jave/...      Your test code
+```
+
+1. The [Java naming conventions](http://java.about.com/od/javasyntax/a/nameconventions.htm) should be used for source files.
 1. Every Java file must have a license header. You can run ```mvn license:format``` on the root of the repo to automatically add missing headers.
-1. Every class, interface and enumeration should have JavaDoc describing its purpose and usage.
-1. Every class, interface and enumeration must have an `@author` tag in its JavaDoc for every author that wrote a substantial part of the file.
-1. Every constant, field and method with default, protected or public visibility should have JavaDoc (optional, but encouraged for private visibility as well).
-1. Code must be formatted using the provided code formatter and clean up settings. They are set up automatically by the official [IDE setup](ide.html).
+1. Code must be formatted using the "ESH" code formatter (in Eclipse).
+   - This is set up automatically by the official [IDE setup](ide.html)
+   - You can manually import [ESH.xml](ESH.xml) via Eclipse Preferences -> Java -> Code Style -> Formatter.
```

Comment:
Do you now the correct location of ESH xml file, to replace the link with a valid one?

Reply:
There is none yet. I need to upload it. I'm creating a second PR for this, as I'm using the github web interface.